test/java/seedu/addressbook/data/AddressBookTest.java

```diff
@@ -0,0 +1,275 @@
+        AddressBook addressBook = createDefaultAddressBook();
+
+        assertFalse(addressBook.containsTag(tagEconomist));
+        assertFalse(addressBook.containsTag(tagPrizeWinner));
+        addressBook.addPerson(davidElliot);
+        assertTrue(addressBook.containsTag(tagEconomist));
+        assertTrue(addressBook.containsTag(tagPrizeWinner));
+    }
+
+    @Test
+    public void syncTagsWithMasterList_defaultAddressBook_tagEqualityPropertyHolds() throws Exception {
+        AddressBook addressBook = createDefaultAddressBook();
+        UniquePersonList persons = addressBook.getAllPersons();
+        UniqueTagList tagsInPersons = getAllTags(persons);
+        UniqueTagList tags = addressBook.getAllTags();
+        
+        // all person tags are equal to some tag in the master list
+        boolean allPersonTagsValid = true;
+        for (Tag personTag : tagsInPersons) {
+            // particular person tag is equal to some tag in the master list
+            boolean personTagValid = false;
+            for (Tag tag : tags) {
+                personTagValid = personTagValid || (personTag == tag);
+            }
+            allPersonTagsValid = allPersonTagsValid && personTagValid;
+        }
+        
+        assertTrue(allPersonTagsValid);
+    }
+
+    @Test
+    public void addPerson_personNotInList_addsNormally() throws Exception {
+        AddressBook addressBook = createDefaultAddressBook();
+        
+        // add person whose tag is in tag list
+        assertFalse(addressBook.containsPerson(charlieDouglas));
+        addressBook.addPerson(charlieDouglas);
+        assertTrue(addressBook.containsPerson(charlieDouglas));
+        
+        // add person whose tag is not in tag list
+        assertFalse(addressBook.containsPerson(davidElliot));
+        addressBook.addPerson(davidElliot);
+        assertTrue(addressBook.containsPerson(davidElliot));
+    }
+
+    @Test
+    public void addPerson_personAlreadyInList_throwsDuplicatePersonException() throws Exception {
+        AddressBook addressBook = createDefaultAddressBook();
+        
+        thrown.expect(DuplicatePersonException.class);
+        addressBook.addPerson(aliceBetsy);
+    }
+
+    @Test
+    public void addTag_tagNotInList_addsNormally() throws Exception {
+        AddressBook addressBook = createDefaultAddressBook();
+
+        assertFalse(addressBook.containsTag(tagEconomist));
+        addressBook.addTag(tagEconomist);
+        assertTrue(addressBook.containsTag(tagEconomist));
+    }
+
+    @Test
+    public void addTag_tagAlreadyInList_throwsDuplicateTagException() throws Exception {
+        AddressBook addressBook = createDefaultAddressBook();
+        
+        thrown.expect(DuplicateTagException.class);
+        addressBook.addTag(tagMathematician);
+    }
+
+    @Test
+    public void containsPerson() throws Exception {
+        AddressBook addressBook = createDefaultAddressBook();
+
+        assertTrue(addressBook.containsPerson(aliceBetsy));
```

Comment:
Loop over all Persons that is supposed to be in the list?
You should also test this method for empty list.